x-pack/legacy/plugins/reporting/server/routes/lib/authorized_user_pre_routing.ts

```diff
@@ -18,9 +19,10 @@ export type PreRoutingFunction = (
 
 export const authorizedUserPreRoutingFactory = function authorizedUserPreRoutingFn(
   server: ServerFacade,
+  plugins: ReportingSetupDeps,
   logger: Logger
 ) {
-  const getUser = getUserFactory(server, logger);
+  const getUser = getUserFactory(server, plugins, logger);
   const config = server.config();
```

Comment:
NIT: unless `authorizedUserPreRoutingFactory` / `getUserFactory` are going to be using other plugins later, I would specialize the parameter and use the security contract instead of the whole `ReportingSetupDeps` object

Reply:
later, we'll use the `licensing` plugin in this area